components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCConfigDAOImpl.java

```diff
@@ -65,22 +69,26 @@ public String getGlobalPolicyAlgorithmName() {
      * @throws EntitlementException throws if fails.
      */
     @Override
-    public void setGlobalPolicyAlgorithm(String policyCombiningAlgorithm) throws EntitlementException {
+    public void addGlobalPolicyAlgorithm(String policyCombiningAlgorithm) throws EntitlementException {
 
         int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
 
         // Check the existence of the algorithm
-        String algorithm = getPolicyCombiningAlgorithm();
+        String algorithm = getPolicyCombiningAlgorithm(tenantId);
         if (StringUtils.isBlank(algorithm)) {
-            setPolicyCombiningAlgorithm(policyCombiningAlgorithm, tenantId);
+            insertPolicyCombiningAlgorithm(policyCombiningAlgorithm, tenantId);
```

Comment:
`addPolicyCombiningAlgorithm` would be a better name.

Reply:
going with the naming convention i have used: 
`get`, `add`, `update`, `remove` for public methods
`get`, `insert`, `update`, `delete` for pure dao methods

Reply:
Ack